### DISCUSSION TITLE
Voxxrin 3 Monetization

### DISCUSSION THREAD

**Discussion by fcamblor** [voxxrin/voxxrin3#58] [Announcements]
### TLDR;

- Voxxrin **will remain OSS**
- In the future (not before 2024), there might be **a Voxxrin offering** for **Conference Organizers**, to be able to declare their event on Voxxrin
- There will be Free Access **for the first event** you organize, so that **you can test Voxxrin at least once** prior to **paying for it**
- There will be a **Free Tier** for **very small conferences** (_those with less than 16 talks in the day, like the JUG Summer Camp_)
- Prices **will remain reasonable** for small conferences with small audience. As an idea, Voxxrin should cost you **no more** than **€1 per day per conference participant**.
- Voxxrin will remain **free of charge** for conference's **attendees** (including **speakers**)
- Potential **contributors** to the codebase need to be aware that their work might be repackaged into **paid product** in the terms of the AGPL v3 license (this license protects the project from deployments of **closed source** instances of Voxxrin)
- At the moment, Voxxrin is **in closed beta** with a limited list of supported conferences. If you're intested into adding your event to the list, please drop us an email to contact 'at' voxxr.in telling us how we could retrieve your agenda (API ? Website to crawl ?)

### Why do I hope Voxxrin v3 will last while v1 & v2 have gradually disappeared?

I can't predict the future, but I'd say that what didn't work in previous Voxxrin versions was about **commitment & motivation on the long run** of contributors (myself, @fcamblor, included).

For example, the **infrastructure costs** were borne by the project maintainers, who also volunteered their time to **maintain the solution in operational conditions**.

We all know that it's a different story to **provide source code** (_more or less "once for all"_) vs **run & maintain it in operational conditions over time.**

By living in a capitalistic society where time is money, it is very complicated to find people who can dedicate free time on the long run, without monetizing it (we all have bills to pay at the end of the month).

It's at least the way the main contributors of Voxxrin v3 (@robinlopez & @fcamblor) are thinking: if conference organizers want a good level of support, bug fixes, improvements (aka, **commitment**), it is necessary to play with real money (not only goodwill).

It's an important move I'd like to make in order to make the project more sustainable, but also to be transparent with future contributors to the project: **there will be monetization of Voxxrin v3 runtime usages at some point in the future**, by a legal entity operated by either @fcamblor or @robinlopez (or both).

### Are there implication on the licensing of the code ?

First and foremost, and for the avoidance of any doubt, **this statement doesn't call into question the opensource nature of the code**.

To make it bold: **the source code of the project will remain OPEN**

Allowing:
1/ anyone to **run the code** locally and easily contribute bug fixes or new features to the project.
2/ anyone to **deploy the app** wherever they want, as long as they respect the AGPL license, if they want to operate the app by themselves without contributing financially to the project.

A few comments nonetheless:
- AGPL license only forbids deploying a **modified** version of the Voxxrin code **without the modifications being open sourced**.
  It means that if you deploy Voxxrin code **without modifying it**, or you deploy **modified** Voxxrin code, **_after having open sourced it_**, you're perfectly within your rights.

- The "spirit of sharing" that Voxxrin project has been promoting since the beginning of its 11y+ existence must not be compromised by these statements.
   As an external contributor to the project, please don't think that we will try to make money from your contributions.
   The idea behind monetization is to find a source of funding **for the production aspects**, but also to guarantee the **contributions of the project's main commiters over the long term**.
   I (@fcamblor) don't rule out sponsorship in the future, if the project reaches a certain break-even point covering the initial investments of the main contributors (@robinlopez and @fcamblor). 
    But we're not there yet unfortunately.

For an order of magnitude of initial investment, since the beginning of the project in Apr 2023, @fcamblor tracked ~360h spent already on the project. That's 45+ days (holidays, nights and week end) spent on the product.
At a ~500€/d rate (which is pretty common for his 17+ years of experience in IT), this would represent a 22.5k€ initial investment on the project so far (and we're far from the end if you want my opinion).

Once the project will be profitable (let's say these initial investments have been repaid), then we'll consider retributing external contributors to the project through sponsorship.

### So, what's next ?

In the 6-9 upcoming months (from Oct 2023 to March-June 2024), Voxxrin v3 will be in a **Closed Beta** mode which will allow : 
- To **reconnect** with the different conferences
- Build **new features**
- And **setup the legal structure** aimed at **operating** Voxxrin v3 **production conditions & data**

After this period of time: **we should open paying tiers**, but as a conference organizer **don't be afraid of it**:
- Usage of Voxxrin 3 **for the first time** on your event will remain free (so that you can see how powerful Voxxrin is, without fearing any mismatch between your needs and our implementation).
- There will likely be a **Free tier** for "very small conferences"
- Prices will be reasonable, and **will depend on the size of the conference**: small conferences should not pay much while we will ask for a more expensive fee on bigger conferences.
  _I have been BDX I/O Conference organizer myself and I know how much it can be stressful, particularly for first time conferences, to find extra budget like the one used for an app._
  To be 100% transparent, my objective is to have a cost **never higher than 1€ per day per attendee** for you.
- Monetizing the app is not intending to make us (@robinlopez and @fcamblor) Billionaires 🤑 but **to cover infrastructure costs** + **motivating us to finding the (day) time to improve the product on the long run** (which will be beneficial to all conferences using it). We give ourselves a period of 1-2 years **to find a market fit** for the app: be aware that if no market fit is found within this time window, it's quite possible that **enthusiasm/commitment on the project will drop drastically**.

As a final note, be aware that both @robinlopez and myself (@fcamblor) will be able to dedicate "week time" to the project (for example, @fcamblor is working at a 80% rate for 4SH, and @robinlopez has a freelancer company) on a regular basis, and for the long run.

**Comment by nicmarti** [comment]
Congratulations for the move, and Devoxx France will be happy to contribute and to help.
